server/bin/pbench-server-prep-shim-002.py

```diff
@@ -0,0 +1,246 @@
+        logger.error("Failed: {} does not exist, or is not a directory", qdir)
+        return 2
+
+    receive_dir_prefix = config.get("pbench-server", "pbench-receive-dir-prefix")
+    if len(receive_dir_prefix) == 0:
+        logger.error("Failed: getconf.py pbench-receive-dir-prefix pbench-server")
+        return 2
+
+    receive_dir = f"{receive_dir_prefix}-002"
+    if not os.path.isdir(receive_dir):
+        logger.error("Failed: {} does not exist, or is not a directory", receive_dir)
+        return 2
+
+    qdir_md5 = os.path.join(qdir, "md5-002")
+    if not os.path.exists(qdir_md5):
+        try:
+            os.makedirs(qdir_md5)
+        except FileExistsError:
+            # directory already exists, ignore
+            pass
+        except Exception:
+            logger.exception(
+                "os.mkdir: Unable to create quarantine destination directory: {}",
+                qdir_md5,
+            )
+            return 1
+
+    duplicates = os.path.join(qdir, "duplicates-002")
+    if not os.path.exists(duplicates):
+        try:
+            os.makedirs(duplicates)
+        except FileExistsError:
+            # directory already exists, ignore
+            pass
+        except Exception:
+            logger.exception(
+                "os.mkdir: Unable to create duplicates destination directory: {}",
+                duplicates,
+            )
+            return 1
+
+    # The following directory holds tarballs that are quarantined because
+    # of operational errors on the server. They should be retried after
+    # the problem is fixed: basically, move them back into the reception
+    # area for 002 agents and wait.
+
+    errors = os.path.join(qdir, "errors-002")
+    if not os.path.exists(errors):
+        try:
+            os.makedirs(errors)
+        except FileExistsError:
+            # directory already exists, ignore
+            pass
+        except Exception:
+            logger.exception(
+                "os.mkdir: Unable to create errors destination directory: {}", errors
+            )
+            return 1
+
+    list_check = list()
+    receive_dir_check = glob.iglob(os.path.join(receive_dir, "**"), recursive=True)
+    for tar in receive_dir_check:
+        if tar.endswith(".tar.xz.md5"):
+            list_check.append(tar)
+
+    list_check.sort()
+    # logger.info(config.TS)
+
+    ntotal = ntbs = nerrs = nquarantined = ndups = 0
+
+    for tbmd5 in list_check:
+        ntotal += 1
+        tb = tbmd5[0:-4]
+        tbdir = os.path.dirname(tb)
+        result_name = os.path.basename(tb)
+
+        controller = os.path.basename(tbdir)
+        dest = os.path.join(config.ARCHIVE, controller)
+
+        if (
+            os.path.isfile(os.path.join(dest, result_name))
+            and os.path.isfile(os.path.join(dest, os.path.basename(tbmd5)))
+            and os.path.exists(os.path.join(dest, result_name))
+            and os.path.exists(os.path.join(dest, os.path.basename(tbmd5)))
```

Comment:
Do we need to both check with `isfile()` and with `exists`?  If we use Pathlib, we might be able to leverage that class's methods to make this a bit cleaner.